packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/transfer/database/database_io.dart

```diff
@@ -51,16 +52,31 @@ class TransferDatabase extends _$TransferDatabase {
               dateTime.toIso8601String(),
             ),
           ))
+        .map(
+          (e) => TransferData(
+            objectKey: e.objectKey,
+            uploadId: e.uploadId,
+            createdAt: e.createdAt,
+          ),
+        )
         .get();
   }
 
-  /// Inserts [entry], a [TransferRecordsCompanion], to the database.
-  Future<int> insertTransferRecord(TransferRecordsCompanion entry) {
-    return into(transferRecords).insert(entry);
+  /// {@macro amplify_storage_s3_dart.transfer_database_insert}
+  @override
+  Future<String> insertTransferRecord(TransferData data) {
+    final entry = TransferRecordsCompanion.insert(
+      uploadId: data.uploadId,
+      objectKey: data.objectKey,
+      createdAt: data.createdAt,
+    );
+    return into(transferRecords)
+        .insert(entry)
+        .then((value) => value.toString());
   }
 
-  /// Deletes [TransferRecords] from the database where
-  /// [TransferRecords.uploadId] is equal to [uploadId].
+  /// {@macro amplify_storage_s3_dart.transfer_database_delete}
```

Comment:
Overrides don't need doc comments since they'll inherit them from the super method.